src/VisualLinter/Helpers/StylelintHelper.cs

```diff
@@ -0,0 +1,292 @@
+
+namespace jwldnr.VisualLinter.Helpers
+{
+    internal static class StylelintHelper
+    {
+        private const string ExecutableName = "stylelint.cmd";
+        private const string VariableName = "stylelint";
+        private static readonly IVisualLinterOptions Options;
+
+        private static readonly string[] SupportedConfigs =
+        {
+            ".stylelintrc.js",
+            ".stylelintrc.yaml",
+            ".stylelintrc.yml",
+            ".stylelintrc.json",
+            ".stylelintrc"
+        };
+
+        static StylelintHelper()
+        {
+            Options = ServiceProvider.GlobalProvider.GetMefService<IVisualLinterOptions>() ??
+                throw new Exception("exception: helper unable to retrieve options");
+        }
+
+        internal static string GetArguments(string directoryPath)
+        {
+            var arguments = new Dictionary<string, string> { { "formatter", "json" } };
+
+            var configPath = GetConfigPath(directoryPath);
+            arguments.Add("config", configPath);
+
+            var ignorePath = GetIgnorePath(directoryPath);
+            if (string.IsNullOrEmpty(ignorePath))
+                return FormatArguments(arguments);
+
+            arguments.Add("ignore-path", ignorePath);
+            return FormatArguments(arguments);
+        }
+
+        internal static string GetConfigPath(string directoryPath)
+        {
+            OutputWindowHelper.DebugLine($"ShouldOverrideStylelintConfigPath: {Options.ShouldOverrideStylelintConfigPath}");
+
+            // resolve override stylelint config path
+            if (Options.ShouldOverrideStylelintConfigPath)
+            {
+                var overridePath = GetOverrideStylelintConfigPath() ??
+                    throw new Exception("exception: option 'Override .stylelintignore path' is set to true-- but no path is set");
+
+                OutputWindowHelper.DebugLine($"using override stylelint config @ {overridePath}");
+
+                return ValidateOverridePath(overridePath);
+            }
+
+            OutputWindowHelper.DebugLine($"UsePersonalStylelintConfig: {Options.UsePersonalStylelintConfig}");
+
+            // resolve personal config path
+            if (Options.UsePersonalStylelintConfig)
+            {
+                var globalPath = GetPersonalConfigPath() ??
+                    throw new Exception("exception: no personal stylelint config found");
+
+                OutputWindowHelper.DebugLine($"using personal stylelint config @ {globalPath}");
+
+                return globalPath;
+            }
+
+            // resolve local stylelint config
+            var localPath = GetLocalConfigPath(directoryPath) ??
+                throw new Exception("exception: no local stylelint config found");
+
+            OutputWindowHelper.DebugLine($"using local stylelint config @ {localPath}");
+
+            return localPath;
+        }
+
+        internal static string GetStylelintPath(string directoryPath)
```

Comment:
Should be renamed to `GetExecutablePath` and exist in both `EslintHelper.cs` and `StylelintHelper.cs`, however methods used by this should be modified.
For example method `GetGlobalStylelintPath` could be moved to `LinterHelper.cs` and be renamed `GetGlobalPath` and be modified to accept a `string variableName` parameter.